### ISSUE TITLE
Improve guidelines when reviewing PRs with backward incompatible changes

### ISSUE THREAD

**Issue by Mariatta** [python/devguide#655]
**Describe the enhancement or feature you'd like**

In the [python-dev](https://mail.python.org/archives/list/python-dev@python.org/message/O3T7SK3BGMFWMLCQXDODZJSBL42AUWTR/) thread, Victor expressed frustration
over lack of documentation of how to port/migrate a function
that was removed.

It would be great that going forward, when we
introduce backward incompatible changes, the documentation
on how to migrate is also included alongside
the feature implementation.

When reviewing pull requests with such changes, e.g.
when something is being deprecated or removed, reviewers
should take notice and potentially block the pull
request from getting merged, until the documentation
gets updated too.

Currently we don't have such guideline or rule. It would
be great to start recommending this practise.

We can probably add this to:

https://devguide.python.org/pullrequest/#how-to-review-a-pull-request

and/or

https://devguide.python.org/committing/#is-the-pr-ready-to-be-accepted

Thanks.

**Comment by jablonskidev** [comment]
@Mariatta I'm already working on the [Accepting Pull Requests](https://devguide.python.org/committing/) page, and it's being reviewed by @brettcannon. I can add a bit about making sure that documentation is up to date as well.

**Comment by jablonskidev** [comment]
@Mariatta Can this issue be closed? Here's a screenshot from the [Accepting Pull Requests](https://devguide.python.org/committing/) page:

![devguide](https://user-images.githubusercontent.com/36643503/107666981-fc833d00-6c43-11eb-9f8a-c1bce02d1001.png)

Would it be helpful if I went through the list of open issues for the devguide and flagged ones that seem to have been addressed already or may be duplicates? It could help new contributors get situated more easily and see a clearer path towards making their first contribution.
